src/ui/buttons/buttons.jsx

```diff
@@ -150,6 +152,28 @@ export function validateButtonProps(props: ButtonPropsInputs) {
   normalizeButtonProps(props);
 }
 
+function calculateMessagePosition({ message, showPoweredBy, layout }): string {
+  if (!message) {
+    return "none";
+  }
+  const { position } = message;
+
+  if (showPoweredBy && position === MESSAGE_POSITION.BOTTOM) {
+    throw new ValidationError(
+      "Message position must be 'top' when Debit and/or Credit Card button is present"
+    );
+  }
+
+  if (
+    showPoweredBy ||
+    position === MESSAGE_POSITION.TOP ||
+    (layout === BUTTON_LAYOUT.VERTICAL && !position)
+  ) {
+    return MESSAGE_POSITION.TOP;
+  }
+  return MESSAGE_POSITION.BOTTOM;
+}
```

Comment:
option to move this function `ui/buttons/message.jsx` (I did not see it used elsewhere and/or reference variables in the parent scope)

Reply:
I like the idea of decluttering `buttons.jsx`; I'd like to suggest `ui/buttons/util.js` as another good place to consider moving that function.  Thoughts, preference?

Reply:
I think `ui/buttons/util.js` would work well! 🚀